tests/layers/tuple_matchers/threshold_tuple_matcher_test.py

```diff
@@ -0,0 +1,140 @@
+        calculated_mask_include = K.eval(
+                ThresholdTupleMatcher({"type": "cosine_similarity"},
+                                      self.num_hidden_layers,
+                                      self.hidden_layer_width,
+                                      hidden_layer_activation=self.hidden_layer_activation)
+                .compute_mask([tuple1, tuple2], [mask1, mask2]))
+        assert_array_almost_equal(calculated_mask_include, numpy.array([[1]], dtype='int32'))
+        assert calculated_mask_include.shape == (1, 1,)
+
+    def test_handles_input_masks_correctly(self):
+        # Custom initialization for this test to account for rounding in the cosine similarity.
+        # pylint: disable=unused-variable,unused-argument
+        def custom_initialization(shape, name=None):
+            return K.ones(shape) * 0.999
+
+        num_slots = 3
+        num_words = 5
+        embed_dimension = 4
+        tuple1_word_input = Input(shape=(num_slots, num_words), dtype='int32', name="input_tuple1")
+        tuple2_word_input = Input(shape=(num_slots, num_words), dtype='int32', name="input_tuple2")
+
+        embedding = TimeDistributedEmbedding(10, embed_dimension, mask_zero=True)
+        embedded_masked_tuple1 = embedding(tuple1_word_input)
+        embedded_masked_tuple2 = embedding(tuple2_word_input)
+
+        match_layer = ThresholdTupleMatcher({"type": "cosine_similarity"},
+                                            self.num_hidden_layers,
+                                            self.hidden_layer_width,
+                                            hidden_layer_activation=self.hidden_layer_activation)
+        output = match_layer([embedded_masked_tuple1, embedded_masked_tuple2])
+        mask_model = Model([tuple1_word_input, tuple2_word_input], output)
+
+        tuple1_words = numpy.random.randint(0, 10, (1, num_slots, num_words))
+        tuple2_words = numpy.zeros((1, num_slots, num_words))
+        # Add a set of matching zeros to slot 1 in tuple1 -- but shouldn't "match" because it's padded
+        tuple1_words[:, 1, :] = numpy.zeros(num_words)
+
+        # Get the initial weights for use in testing
+        layer_nn = match_layer.hidden_layer_weights
+
+        # Testing general unmasked case.
+        desired_overlap = K.variable(numpy.asarray([[0, 0, 0]]))
+        # Desired_overlap gets fed into the inner NN.
+        neural_network_feed_forward = apply_feed_forward(desired_overlap, layer_nn,
+                                                         activations.get(match_layer.hidden_layer_activation))
+        # Apply the final activation function.
+        desired_result = activations.get(match_layer.final_activation)(K.dot(neural_network_feed_forward,
+                                                                             match_layer.score_layer))
+        result = mask_model.predict([tuple1_words, tuple2_words])
+        assert_array_almost_equal(result, K.eval(desired_result))
```

Comment:
Don't you want the result to be masked here?  So it doesn't really matter what the result is, and you should just check that it's masked.  You can do that using the `OutputMask` layer (something like what's done here: https://github.com/allenai/deep_qa/blob/c9a013993c6f15f1a0d83ec68b726669f38cf543/tests/layers/vector_matrix_split_test.py#L53-L62).